internal/datasource/maven_registry.go

```diff
@@ -223,7 +223,8 @@ func (m *MavenRegistryAPIClient) get(ctx context.Context, auth *HTTPAuthenticati
 			return response{}, fmt.Errorf("%w: Maven registry query status: %d", errAPIFailed, resp.StatusCode)
 		}
 
-		if b, err := io.ReadAll(resp.Body); err == nil {
+		b, err := io.ReadAll(resp.Body)
```

Comment:
I think this whole thing should just be inverted right, the last line should be the valid response, and all the branches are the errors.

Reply:
done